### ISSUE TITLE
Allow configuring local and distributed agents with custom loggers

### ISSUE THREAD

**Issue by nikolamomchev** [deepmind/acme#171]
All agents should allow setting a custom loggers for actors and learners.

This has been requested multiple times. See #156 #133 #166 and #164

**Comment by wookayin** [comment]
@nikolamomchev Any updates on this? It's been more than 3 months and still there seems no progress on this. I thought #164, #166 is an easy, "straightforward" fix (following what already had existed), but you seemed to be concerned about backward compatibility and for the sake of more general interface.

**Comment by nikolamomchev** [comment]
@wookayin sorry for the delay on this. We are making slow progress on this direction but it will probably take a bit longer until this is finalised.

This commit was a small step in this direction:
https://github.com/deepmind/acme/commit/496042df0db5d8f5e1b73146176ecc84c4f5c7ec

**Comment by wookayin** [comment]
@qstanczyk Thanks for the update. I checked out the new example, but I am not sure the new ExperimentConfig and Experiment API is better and really address the feature request here. I see some design issues here:

* The `acme.jax.experiments` design, seems overlap with `LocalLayout`. I feel the role are basically the same (instantiating learner, actor, etc. components akin to distributed nodes builder). In this case, is `LocalLayout` going to be deprecated (like distributed_layout was deprecated) and shouldn't be used? All that said, the `LocalLayout` API still exists,  but this is **still not configurable** in logging behaviors.

* What I don't like about `acme.jax.experiments` is, it looks like too **monolithic**. All the procedures of creating networks/learner/actor/logger/etc. and running them through train-eval loop is encapsulated in a **single function** --- this can be handy in some cases if things can be specified as `ExperimentConfig`, but it doesn't seem as flexible as object-oriented APIs (LocalLayout). For instance, you cannot override or customize a specific part of it **that is not exposed** as a configuration point in `ExperimentConfig`.

**Comment by qstanczyk** [comment]
Thanks for detailed analysis of the new design. Some responses to your questions:
* `acme.jax.experiments` is supposed to replace layouts and agents. This reduces number of layers new users need to understand to start playing with Acme. It also puts the main execution loop closer to the entry point, hopefully making code understandability easier for newcomers.

* Re `learner_logger` - in case of `ExperimentConfig` provides a logger factory which is used for any component type, so in order to specialise logger's behaviour it needs to be done in one central place only.

* In case of AIL, RND etc. - by default each learner will get its own logger instance from the logger's factory. If it is needed to have a different behaviour between those loggers is is possible to vary between the learners based on the label and construct a different logger for each of them.

* The idea is that if you want to override the logic of the training loop it should be copied. It is now close to the entry point, so it should be easy to swap one for the other.

**Comment by wookayin** [comment]
@qstanczyk, Thank you so much for your detailed response and help! Yes, I agree that the new experiments API is better and certainly has the benefits that you mentioned. I was just confused whether LocalAgent implementations (e.g., `SAC`, `PPO` classes) are deprecated or not: I should be migrating my code to the new pattern to make loggers fully configurable. Looking forward to seeing them deprecated and documented in following CLs.

I think we could later have some further improvements to the API so that the user code becomes more concise and less verbose, including some refactorings for `run_experiment` to make it more reusable. (Although I feel like is what LocalAgent was doing...)  Let me try to play around with them and let you know if I have further suggestions..

**Comment by qstanczyk** [comment]
Perfect, thanks.
